src/twisted/conch/test/test_cftp.py

```diff
@@ -1436,6 +1436,7 @@ def _cbCheckResult(res):
 @skipIf(skipTests, "don't run w/o spawnProcess or cryptography")
 @skipIf(not which("ssh"), "no ssh command-line client available")
 @skipIf(not which("sftp"), "no sftp command-line client available")
+@skipIf(not hasDsa, "needs ssh supporting dsa")
```

Comment:
IMO this skips too much.  Most of these tests still make conceptual sense to run with an OpenSSH that doesn't support DSA - we just need to run them in a way that doesn't rely on DSA.  I haven't checked in detail, but I'd have thought most of them could just use an RSA key instead.

(We may find that we can do without the `hasDsa` helper, in which case sorry for the misdirection!  I assumed that there were tests that intrinsically relied on DSA support, but it's not completely clear that's the case.)

Reply:
this one actually seems to skip only `twisted.conch.test.test_cftp.OurServerSftpClientTests.test_extendedAttributes`, which was failing.